File: 9506/code.c
#include <stdio.h>
#include <string.h>
#include <math.h>


int main() {

   int num;
   int sum;
   int arr[1000];
   int k;

   while(1){

   scanf("%d", &num);

   if(num == -1){

    break;
   }

   sum = 0;
   k = 0;

   for(int i=1; i<num; i++){

    if(num % i ==0){

        sum = sum +i;
        arr[k] = i;
        k++;
    }

   }

   if(num==sum){

   printf("%d = %d", num, arr[0]);
   for(int i=1; i<k;i++){

            printf("+%d", arr[i]);

   }

   printf("\n");

}
    else if(num!=sum){

        printf("%d is NOT perfect", num);
    }
}
}

File: 9506/code.py
while True :
    
    sum = 0
    arr = []
    num = int(input())
    
    if num == -1 :

        break

    for i in range(1, num) :

        if num%i == 0 :

            sum = sum + i
            arr.append(i)
    if sum == num :
        print(num, "=", arr[0], end="")
        for i in range(1, len(arr)) :
            print(" +", arr[i], end ="")
        print()
    else :
        print(num, "is NOT perfect.", end="")
    
        print()
        
        

    
    
